addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/handler/AirVisualNodeHandler.java

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.airvisualnode.handler;
+
+import static org.openhab.binding.airvisualnode.AirVisualNodeBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.nio.charset.StandardCharsets;
+import java.util.Calendar;
+import java.util.TimeZone;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.airvisualnode.internal.config.AirVisualNodeConfig;
+import org.openhab.binding.airvisualnode.internal.json.NodeData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import jcifs.smb.NtlmPasswordAuthentication;
+import jcifs.smb.SmbFile;
+import jcifs.smb.SmbFileInputStream;
+
+/**
+ * The {@link AirVisualNodeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Victor Antonovich - Initial contribution
+ */
+public class AirVisualNodeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AirVisualNodeHandler.class);
+
+    public static final String NODE_JSON_FILE = "latest_config_measurements.json";
+
+    private final Gson gson;
+
+    private ScheduledFuture<?> pollFuture;
+
+    private long refreshInterval;
+
+    private String nodeAddress;
+
+    private String nodeUsername;
+
+    private String nodePassword;
+
+    private String nodeShareName;
+
+    private NodeData nodeData;
+
+    public AirVisualNodeHandler(@NonNull Thing thing) {
+        super(thing);
```

Comment:
Please do not call `super` here because it sets your `thing` to ONLINE right way!

Reply:
Umm, looks like [BaseThingHandler constructor](https://github.com/eclipse/smarthome/blob/8d0856f424ae8115f1ca98d09c806a12a1827f47/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandler.java#L112) doesn't sets `thing` to ONLINE mode, isn't it? Moreover, Eclipse complains about default constructor absence if I remove this `super` call.

Reply:
Sorry for the confusion, I took the wrong exist O_o I was thinking of `initialize` which sets the `Thing` to ONLINE here https://github.com/eclipse/smarthome/blob/8d0856f424ae8115f1ca98d09c806a12a1827f47/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandler.java#L242